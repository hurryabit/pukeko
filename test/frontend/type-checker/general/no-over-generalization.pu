data Bool =
  | False
  | True
f :: Int
f =
  let g x =
    let h y = x in
    if h True then 0 else h False
  in
  0
