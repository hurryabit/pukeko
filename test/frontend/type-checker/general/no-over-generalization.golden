test/frontend/type-checker/general/no-over-generalization.pu:6:5: mismatching types Int and Bool
